src/daft-core/src/array/ops/utf8.rs

```diff
@@ -937,7 +937,7 @@ impl Utf8Array {
     pub fn to_datetime(&self, format: &str, timezone: Option<&str>) -> DaftResult<TimestampArray> {
         let len = self.len();
         let self_iter = self.as_arrow().iter();
-        let timeunit = crate::datatypes::utils::infer_timeunit_from_format_string(format);
+        let timeunit = daft_schema::time_unit::infer_timeunit_from_format_string(format);
```

Comment:
The rule of thumb im going with is anything that is used in a public API of that module.

Reply:
Also the prelude is an optional opt-in, it's not automatically imported.

Reply:
> The rule of thumb im going with is anything that is used in a public API of that module.

Isn't that just `use daft_core::*`? The `*::prelude::*` pattern feels more like "please import all the stuff I absolutely will need to work with your crate"

For e.g. in `pyo3` I think it's all the pyclass and pymethod stuff, but for more specific things we still do separate imports. 

https://www.lurklurk.org/effective-rust/wildcard.html?highlight=prelude#item-23-avoid-wildcard-imports

> However, there's another common exception where wildcard imports make sense. Some crates have a convention that common items for the crate are re-exported from a prelude module, which is explicitly intended to be wildcard imported:
> ```use thing::prelude::*;```
> Although in theory the same concerns apply in this case, in practice such a prelude module is likely to be carefully curated, and higher convenience may outweigh a small risk of future problems.

Reply:
You can take a look at [datafusion-cores's prelude](https://github.com/apache/datafusion/blob/main/datafusion/core/src/prelude.rs)

Also happy to put things within a namespace under prelude.

Also looks like pyo3 puts a ton under their prelude (all the methods and traits)
https://github.com/PyO3/pyo3/blob/main/src/prelude.rs

https://github.com/PyO3/pyo3/blob/main/src/prelude.rs

Reply:
also `use daft_core::*` would just all the modules that are public which is different behavior.

instead of `Series, DataArray, ListArray` it would do the modules `series`, `array` and `datatypes` which is not what we want.